src/rgw/driver/sfs/object.cc

```diff
@@ -282,26 +287,98 @@ void SFSObject::gen_rand_obj_instance_name() {
   state.obj.key.set_instance(buf);
 }
 
-int SFSObject::get_obj_attrs(
-  optional_yield y,
-  const DoutPrefixProvider *dpp,
-  rgw_obj *target_obj
-) {
-  lsfs_dout(dpp, 10) << " TODO" << dendl;
+/*
+  This should be intended as a mere fetch of object's attributes.
+  The caller will likely call get_attrs() after get_obj_attrs().
+  This is how we have interpreted this call.
+  The average usage of this seems to suggest that one first calls
+  get_obj_attrs() and then it uses get_attrs().
+  Doubts remain that this could be entirely correct for all cases;
+  so for the time being, we leave target_obj empty.
+  We rely on the fact that, if in the future something could potentially fail
+  because target_obj is left empty, that fail will be explicit.
+*/
```

Comment:
Hard to grok. Can we make this more concise? 
My interpretation is:
1. We expect get_attr calls after get_obj_attrs calls. Can we assert this in code?
2. We don't set target_obj. If a caller uses that, boom

Would a dummy / sentinel target_obj make sense?

Reply:
Please, let's always take in consideration that we are not the original designers of the abstract code, therefore we cannot always argue 100% precisely the original semantics as was originally thought by original designers. 
The process of interpretation is by definition uncertain.
Accepting to work with a codebase that was designed by others necessarily leads to compromises; also in terms of interface semantics guessing.
In my opinion an assured boom in an uncertain scenario derived by a lack of knowledge is better than an hiding of that.
Hiding something that is uncertain, IMO is worst that explicitly make it fail.

Reply:
@irq0 I think the requirements to asserting on "get_attrs after get_obj_attrs" would be non-trivial and potentially for little to no gain.

I think a more interesting assert would be on `target_obj`. From the call stack, do we know whether it's expected to be populated when the function is called?

Reply:
Since the signature (RADOS BE) of `get_obj_attrs` is:

```
virtual int get_obj_attrs(optional_yield y, const DoutPrefixProvider* dpp, rgw_obj* target_obj = NULL) override;
```

We should have no doubts that this function can be called with 2 or 3 args.
It's hard to say if there is a rule here, we should check every call to this function to check if, when `target_obj` is not `NULL`, this is an empty object, or instead is a dirty one.

Reply:
> @irq0 I think the requirements to asserting on "get_attrs after get_obj_attrs" would be non-trivial and potentially for little to no gain.

ack